lib/unicorn_wrangler.rb

```diff
@@ -39,9 +61,26 @@ def perform_request
     ensure
       @handlers.each { |handler| handler.call(@requests, @request_time) }
     end
+
+    def perform_hook(name)
+      if hook = @hooks[name]
+        hook.call
+      end
+    end
   end
 
   module UnicornExtension
+    # call our hook and the users hook since only a single hook can be configured at a time
+    # we need to call super so the @<hook> variables get set and unset properly in after_fork to not leak memory
+    [:after_fork, :before_fork].each do |hook|
+      define_method("#{hook}=") do |value|
+        super(->(*args) do
```

Comment:
that’s some black ruby magic right there

Reply:
yeah looks kinda scary :D 
when configuration tries to store a block I wrap it in another block that also executes our callback